aws/resource_aws_cloudfront_function.go

```diff
@@ -0,0 +1,303 @@
+			},
+			"publish": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"stage": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// resourceAwsCloudFrontFunction maps to:
+// CreateFunction in the API / SDK
+func resourceAwsCloudFrontFunctionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	functionName := d.Get("name").(string)
+
+	log.Printf("[DEBUG] Creating Lambda Function %s", functionName)
+
+	params := &cloudfront.CreateFunctionInput{
+		FunctionCode: []byte(d.Get("code").(string)),
+		FunctionConfig: &cloudfront.FunctionConfig{
+			Comment: aws.String(d.Get("comment").(string)),
+			Runtime: aws.String(d.Get("runtime").(string)),
+		},
+		Name: aws.String(functionName),
+	}
+
+	CreateFunctionOutput, err := conn.CreateFunction(params)
+
+	if err != nil {
+		return fmt.Errorf("error creating Cloudfront Function: %w", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	d.Set("version", CreateFunctionOutput.ETag)
+	d.Set("arn", CreateFunctionOutput.FunctionSummary.FunctionMetadata.FunctionARN)
+	d.Set("last_modified", CreateFunctionOutput.FunctionSummary.FunctionMetadata.LastModifiedTime.Format(time.RFC3339))
+	d.Set("stage", CreateFunctionOutput.FunctionSummary.FunctionMetadata.Stage)
+	d.Set("status", CreateFunctionOutput.FunctionSummary.Status)
+
+	publish := d.Get("publish").(bool)
+	if publish {
+
+		params := &cloudfront.PublishFunctionInput{
+			Name:    aws.String(d.Get("name").(string)),
+			IfMatch: aws.String(d.Get("version").(string)),
+		}
+
+		log.Printf("[DEBUG] Publishing Cloudfront Function: %s", d.Id())
+
+		PublishFunctionOutput, err := conn.PublishFunction(params)
+		if err != nil {
+			return err
+		}
+		d.Set("status", PublishFunctionOutput.FunctionSummary.Status)
+		d.Set("last_modified", PublishFunctionOutput.FunctionSummary.FunctionMetadata.LastModifiedTime.Format(time.RFC3339))
+	}
+
+	return nil
+}
+
+// resourceAwsCloudFrontFunctionRead maps to:
+// GetFunction in the API / SDK
+func resourceAwsCloudFrontFunctionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	params := &cloudfront.GetFunctionInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Get Cloudfront Function: %s", d.Id())
+
+	GetFunctionOutput, err := conn.GetFunction(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudfront.ErrCodeNoSuchFunctionExists && !d.IsNewResource() {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
```

Comment:
```suggestion
Import `"github.com/hashicorp/aws-sdk-go-base/tfawserr"` and use `tfawserr.ErrCodeEquals`:

```suggestion
	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, cloudfront.ErrCodeNoSuchFunctionExists) {
		log.Printf("[WARN] CloudFront Function (%s) not found, removing from state", d.Id())
		d.SetId("")
		return nil
	}
	
 	if err != nil {
 	 	return fmt.Errorf("error getting CloudFront Function (%s): %w", d.Id(), err)
 	}
```
```